### ISSUE TITLE
nn layer to quantize the sign of a vector?

### ISSUE THREAD

**Issue by andreamigliorati** [torch/nn#1174]
Hi, I'd like to know if there's a predefined **nn.** layer (or maybe a combined sequence of them) that allows me to obtain, given a vector, another vector containing the quantized sign of the values.
In other words, this function: f(x) = 1 if x >= 0 and 0 elsewhere.
I'd like to do that via a nn layer so I could add it to my structure and just forward it to obtain the binary sequences.

Example: to **a** = torch.Tensor({-0.1, 0.9, -0.33, 0.4, 0.5})
should be returned **b** 
> b
0
1
0
1
1
[torch.DoubleTensor of size 5]

Can you help me out? Thanks in advance

**Comment by iamalbert** [comment]
The gradient is 0 everywhere; you cannot train the layers before this.

**Comment by iacolippo** [comment]
As already said, you can't train layers before the one which contains this operation with backpropagation. If you really want this operation you could try to train the network with the Indirect Feedback Alignment scheme.

**Comment by andreamigliorati** [comment]
Thank you very much. Closing this thread.
